doc/connectivity/bluetooth/bluetooth-ctlr-arch.rst

```diff
@@ -0,0 +1,261 @@
+.. _bluetooth-ctlr-arch:
+
+Bluetooth Low Energy Controller
+###############################
+
+Overview
+********
+
+Architecture
+************
+
+Execution Overview
+==================
+
+.. image:: img/ctlr_exec_overview.png
+
+
+Architecture Overview
+=====================
+
+.. image:: img/ctlr_arch_overview.png
+
+
+Scheduling
+**********
+
+.. image:: img/ctlr_sched.png
+
+
+Ticker
+======
+
+.. image:: img/ctlr_sched_ticker.png
+
+
+Upper Link Layer and Lower Link Layer
+=====================================
+
+.. image:: img/ctlr_sched_ull_lll.png
+
+
+Scheduling Variants
+===================
+
+.. image:: img/ctlr_sched_variant.png
+
+
+ULL and LLL Timing
+==================
+
+.. image:: img/ctlr_sched_ull_lll_timing.png
+
+
+Event Handling
+**************
+
+.. image:: img/ctlr_sched_event_handling.png
+
+
+Scheduling Closely Spaced Events
+================================
+
+.. image:: img/ctlr_sched_msc_close_events.png
+
+
+Aborting Active Event
+=====================
+
+.. image:: img/ctlr_sched_msc_event_abort.png
+
+
+Cancelling Pending Event
+========================
+
+.. image:: img/ctlr_sched_msc_event_cancel.png
+
+
+Pre-emption of Active Event
+===========================
+
+.. image:: img/ctlr_sched_msc_event_preempt.png
+
+
+Data Flow
+*********
+
+Transmit Data Flow
+==================
+
+.. image:: img/ctlr_dataflow_tx.png
+
+
+Receive Data Flow
+=================
+
+.. image:: img/ctlr_dataflow_rx.png
+
+
+Execution Priorities
+********************
+
+.. image:: img/ctlr_exec_prio.png
+
+- Event handle (0, 1) < Event preparation (2, 3) < Event/Rx done (4) < Tx
+  request (5) < Role management (6) < Host (7).
+
+- LLL is vendor ISR, ULL is Mayfly ISR concept, Host is kernel thread.
+
+
+Lower Link Layer
+****************
+
+LLL Execution
+=============
+
+.. image:: img/ctlr_exec_lll.png
+
+
+LLL Resume
+----------
+
+.. image:: img/ctlr_exec_lll_resume_top.png
+
+.. image:: img/ctlr_exec_lll_resume_bottom.png
+
+
+Bare metal utilities
+********************
+
+Memory FIFO and Memory Queue
+============================
+
+.. image:: img/ctlr_mfifo_memq.png
+
+
+Legacy Controller
+*****************
+
+.. image:: img/ctlr_legacy.png
+
+Bluetooth Low Energy Controller - Vendor Specific Details
+*********************************************************
+
+Hardware Requirements
+=====================
+
+Nordic Semiconductor
+--------------------
+
+The Nordic Semiconductor Bluetooth Low Energy Controller implementation
+requires the following hardware peripherals.
+
+.. list-table:: SoC Peripheral Use
+   :header-rows: 1
+
+   * - Resource
+     - nRF Peripheral
+     - Instances
```

Comment:
`# instances` or instance count

Reply:
Or even better: remove it if you split the "nRF Peripheral" column as I suggest below

Reply:
Updated to `# instances`

Reply:
For now not adding any more columns